File: algorithms/code/leetcode/lc001_two_sum/LC001TwoSum.Tests.cs
using Xunit;

namespace AlgorithmDesign.code.leetcode.lc001_two_sum
{
    public class SolutionTests
    {
        [Fact]
        public void Test0()
        {
            Assert.Null(new Solution().TwoSum(null, 0));
        }

        [Fact]
        public void Test00()
        {
            Assert.Null(new Solution().TwoSum(new int[] { }, 0));
        }

        [Fact]
        public void Test1()
        {
            Assert.Equal(new[] { 0, 1 }, new Solution().TwoSum(new[] { 2, 7, 11, 15 }, 9));
        }

        [Fact]
        public void Test2()
        {
            Assert.Equal(new[] { 1, 2 }, new Solution().TwoSum(new[] { 3, 2, 4 }, 6));
        }

        [Fact]
        public void Test3()
        {
            Assert.Equal(new[] { 0, 1 }, new Solution().TwoSum(new[] { 3, 3 }, 6));
        }
    }
}

File: algorithms/code/leetcode/lc001_two_sum/LC001TwoSumTests.java
package leetcode.lc001_two_sum;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

public class LC001TwoSumTests {
    @Test
    public void test0() {
        assertNull(new LC001TwoSum().twoSum(null, 0));
    }

    @Test
    public void test00() {
        assertNull(new LC001TwoSum().twoSum(new int[] {}, 0));
    }

    @Test
    public void test1() {
        int[] expected = new int[] { 0, 1 };
        int[] result = new LC001TwoSum().twoSum(new int[] { 2, 7, 11, 15 }, 9);
        assertArrayEquals(expected, result);
    }

    @Test
    public void test2() {
        int[] expected = new int[] { 1, 2 };
        int[] result = new LC001TwoSum().twoSum(new int[] { 3, 2, 4 }, 6);
        assertArrayEquals(expected, result);
    }

    @Test
    public void test3() {
        int[] expected = new int[] { 0, 1 };
        int[] result = new LC001TwoSum().twoSum(new int[] { 3, 3 }, 6);
        assertArrayEquals(expected, result);
    }
}
